config/main.py

```diff
@@ -4098,67 +4084,59 @@ def add_contact(db, contact, contact_email):
 
 
 @contact.command('del')
-@click.argument('contact', metavar='<contact_name>', required=True)
+@click.argument('contact', metavar='<contact>', required=True, nargs=-1)
 @clicommon.pass_db
-def del_contact(db, contact):
-    """ Delete snmp contact name and email """
+def delete_contact(db, contact):
+    """ Delete snmp contact"""
+    config_db = ValidatedConfigDBConnector(db.cfgdb)
+    if isinstance(contact, tuple):
+        contact = " ".join(contact)
     snmp = db.cfgdb.get_table("SNMP")
     try:
-        if not contact in (list(snmp['CONTACT'].keys()))[0]:
-            click.echo("SNMP contact {} is not configured".format(contact))
-            sys.exit(1)
-        else:
-            db.cfgdb.set_entry('SNMP', 'CONTACT', None)
-            click.echo("SNMP contact {} removed from configuration".format(contact))
+        if contact == snmp['CONTACT']['Contact']:
+            try:
+                config_db.set_entry('SNMP', 'CONTACT', None)
+                click.echo("SNMP Contact {} removed from configuration".format(contact))
+            except (ValueError, JsonPatchConflict) as e:
+                ctx = click.get_current_context()
+                ctx.fail("Failed to remove SNMP contact from configuration. Error: {}".format(e))
             try:
                 click.echo("Restarting SNMP service...")
                 clicommon.run_command(['systemctl', 'reset-failed', 'snmp.service'], display_cmd=False)
                 clicommon.run_command(['systemctl', 'restart', 'snmp.service'], display_cmd=False)
             except SystemExit as e:
                 click.echo("Restart service snmp failed with error {}".format(e))
                 raise click.Abort()
+        else:
+            click.echo("SNMP Contact {} does not exist.".format(contact))
+            sys.exit(1)
     except KeyError:
         if "CONTACT" not in snmp.keys():
```

Comment:
>if "CONTACT" not in snmp.keys(): [](http://example.com/codeflow?start=8&length=32)

another possibility is that 'Contact' is not in snmp['CONTACT']. If this happen, the exception is silently swallowed.